gh-135552: Clear weakrefs to types in GC after garbage finalization not before (#135728)
python/cpython: Misc/NEWS.d/next/Core_and_Builtins/2025-06-23-01-07-09.gh-issue-135552.eG9vzP.rst

Split the handling of weak references in the GC if both types and instances

> within a generation are unreachable. This prevent  the clearing  of the

Reviewer: ```suggestion
within a generation are unreachable. This prevent the clearing of the
```

> type's subclasses list too early. This fix a segfault that occurs when

Reviewer: ```suggestion
type's subclasses list too early. This fixes a crash that occurs when
```

IMO, it's better to use word "crash" because "segfault" is platform specific.

Author: Thanks!

> __del__ modifies the base's attributes and tries to access them from self.

Reviewer 2: ```suggestion
:meth:`~object.__del__` modifies the base's attributes and tries to access them from self.
```

Author: Thanks!
